### PR TITLE
Selection plugin: Initial version of selection_to_tcl_list command

### PR DESCRIPTION
**Pull request by tmichalak**
Initial version of the selection_to_tcl_list command.

### CODE REVIEW

**Comment by tmichalak** [comment]
@litghost 
This is just the initial version. I could use some advice on the command's interface that has a chance of being accepted upstream. Maybe Eddie could be helpful here?

File: selection-plugin/selection.cc [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		//   |---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|---v---|
+		log("\n");
+		log("   selection_to_tcl_list selection\n");
+		log("\n");
+		log("Extract the current selection to a Tcl List with selection object names. \n");
+		log("\n");
+	}
+
+	void AddObjectNameToTclList(RTLIL::IdString& module, RTLIL::IdString& object, Tcl_Obj* tcl_list) {
+		std::string name = RTLIL::unescape_id(module) + "/" + RTLIL::unescape_id(object);
```
Comment (litghost) [inline selection-plugin/selection.cc:42]: Why `unescape_id` vs `id2cstr`? See https://github.com/YosysHQ/yosys/blob/ede4b10da8fdbdcff900b93c6c723516901483ff/passes/cmds/select.cc#L1444

Comment (tmichalak) [reply selection-plugin/selection.cc:42]: Because `id2cstr` is to c_str whereas `unescape_id` is to string. Considered `std::string(id2cstr())`.

Comment (litghost) [reply selection-plugin/selection.cc:42]: As long as they provide the same contents, I'm good.  I was just comparing and noticed the difference.

File: selection-plugin/tests/counter/counter.golden.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+1
```
Comment (litghost) [inline selection-plugin/tests/counter/counter.golden.txt:1]: This golden looks broken now, what happened?

Comment (litghost) [reply selection-plugin/tests/counter/counter.golden.txt:1]: I see the problem.  The golden should probably have actually output from `selection_to_tcl_list` and generate a TCL error if the comparision to `-write` fails

Comment (tmichalak) [reply selection-plugin/tests/counter/counter.golden.txt]: Fixed, the reference contains the selection now and the test returns a tcl error upon failure.

**Pull request merged**